src/module_3/evaluation_functions.py

```diff
@@ -0,0 +1,179 @@
+
+import pandas as pd
+import numpy as np
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import  StandardScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.compose import ColumnTransformer
+import category_encoders as ce
+from sklearn.model_selection import StratifiedKFold
+import matplotlib.pyplot as plt
+from datetime import timedelta
+from sklearn.model_selection import PredefinedSplit
+import pyarrow.parquet as pq
+from sklearn import metrics
+from sklearn.metrics import roc_auc_score, confusion_matrix
+
+
+def plot_precision_recall_vs_threshold(clf, X, y):
```

Comment:
This functions should probably not compute `model_probs`. That should be the responsability of an `inference` function if anything. This one, is about plotting precision, recall vs threshold and that's what it should do. It could take as arguments `model_prob` and `y` only.